### PR TITLE
Summon Guns 2016

### PR DESCRIPTION
**Pull request by Incoming5643**
Adds a buttload of guns to summon guns to bring it up to speed for 2016. I intentionally skipped any guns that had enough stopping power to OHKO a spessman from full health in one click.

Adds a gun proc, Unlock(), that automatically swaps whatever pin the gun has for a generic electronic one.

:cl: incoming5643
rscadd: The wizard federation has finally updated their assortment of guns for the summon guns event. Look forward to getting shot in the face with a whole new batch of interesting and rarely seen weaponry!
/:cl:

### CODE REVIEW

**Comment by Bawhoppen** [comment]
eh if its for wiz I guess

also you should add the smaller version of pulse guns, like pulse pistol

**Comment by Anonus** [comment]
Add the xray laser and scatter laser.
Also make the deagle type random from normal deagle, gold-plated deagle, and camo deagle. Deagle.

**Comment by Incoming5643** [comment]
I was firing the scatter weapons and they weren't actually scattering, so I passed.

**Comment by Shapsy** [comment]
You have to use them in hand to switch to scatter mode
Scatter lasers are horribly underrated, we need them more

**Comment by GunHog** [comment]
I would love to give scatter lasers a new sprite and make them research-able, but I feel the community would defecate on me for trying.

**Comment by Iamgoofball** [comment]
do it anyways

File: code/modules/projectiles/gun.dm [hunk 1]
```diff
@@ -428,7 +428,9 @@ obj/item/weapon/gun/proc/newshot()
 
 	process_fire(target, user, 1, params)
 
-
+/obj/item/weapon/gun/proc/unlock() //used in summon guns and as a convience for admins
+	if(pin)	qdel(pin)
```
Comment (Anonus) [inline code/modules/projectiles/gun.dm]: don't do this on the same line it's a code standard

**Comment by KazeEspada** [comment]
Add the 1 shot kill guns, we already have the gyrojet and pulse rifle.

File: code/modules/projectiles/gun.dm (hunk 1 above)
Comment (Incoming5643) [inline code/modules/projectiles/gun.dm]: people have been more anal about "code standards" lately.

I always liked single line if() changes all being on the same line, but whatever.

**Comment by Jordie0608** [comment]
Needs a changelog

File: code/game/gamemodes/wizard/rightandwrong.dm [hunk 2]
```diff
@@ -32,66 +32,90 @@
...
+					G = new /obj/item/weapon/gun/projectile/automatic/wt550(get_turf(H))
+				if("bulldog")
+					G = new /obj/item/weapon/gun/projectile/automatic/shotgun(get_turf(H))
+				if("grenadelauncher")
+					G = new /obj/item/weapon/gun/projectile/revolver/grenadelauncher(get_turf(H))
+				if("goldenrevolver")
+					G = new /obj/item/weapon/gun/projectile/revolver/golden(get_turf(H))
+				if("sniper")
+					G = new /obj/item/weapon/gun/projectile/sniper_rifle(get_turf(H))
+				if("medibeam")
```
Comment (newfren) [inline code/game/gamemodes/wizard/rightandwrong.dm]: Incredibly excited to get the medibeam and survivor status.

Comment (Iamgoofball) [inline code/game/gamemodes/wizard/rightandwrong.dm]: make it a reverse medibeam imo

Comment (GunHog) [inline code/game/gamemodes/wizard/rightandwrong.dm]: We would have it if Dunc did not hate Science >:-(

Comment (Incoming5643) [inline code/game/gamemodes/wizard/rightandwrong.dm]: every roulette needs a booby prize, be thankful I didn't add cap guns instead.

Comment (JJRcop) [inline code/game/gamemodes/wizard/rightandwrong.dm]: Who says you can't still do that :)

File: code/modules/projectiles/gun.dm (hunk 1 above)
Comment (duncathan) [inline code/modules/projectiles/gun.dm]: They're standard for a reason
In this case, it's consistency

File: code/game/gamemodes/wizard/rightandwrong.dm (hunk 2 above)
Comment (duncathan) [inline code/game/gamemodes/wizard/rightandwrong.dm]: @GunHog we can talk when another job gets access to the majority of the cool shit in the game

Science is disproportionately cool; give cool things to other departments too

Comment (Jalleo) [inline code/game/gamemodes/wizard/rightandwrong.dm]: Or rework R&D to be a multi department thing.

Comment (GunHog) [inline code/game/gamemodes/wizard/rightandwrong.dm]: It is just really hard to share with other departments. No easy way to send things off to Cargo for redistribution (Flushing it down disposals will just get it spaced). I WANT to give the CMO a Medgun. I WANT to give Sec better guns. I WANT to give the CE NV Mesons and an experimental welder.

But they do not bother to come by Science to pick them up. I cannot be wasting time ferrying things to other people, that means I do not have time to work on the research tree or help out in my department.

Comment (KazeEspada) [inline code/game/gamemodes/wizard/rightandwrong.dm]: Actually if you wrap something before sending it to cargo, Disposals will reject it from the trash and put it in the mail slot.

**Comment by tkdrg** [comment]
i merged "summon guns 2016" in 2015

**Comment by Jalleo** [comment]
This is how its meant to work.

**Pull request merged**